tests/waveplate_stub_test.py

```diff
@@ -0,0 +1,108 @@
+from pnpq.devices.waveplate_stub import WaveplateStub
+from pnpq.errors import (
+    DeviceDisconnectedError,
+    WaveplateInvalidDegreeError,
+    WaveplateInvalidMotorChannelError,
+    WaveplateInvalidStepsError,
+)
+import pytest
+
+def test_access_without_connection():
+    wp = WaveplateStub()
+    with pytest.raises(DeviceDisconnectedError):
+        wp.identify()
+
+def test_connect():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.identify()
+
+def test_home():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.rotate(90)
+    wp.home()
+    pos = wp.getpos()
+    assert pos == 0
+
+def test_rotate():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.rotate(90)
+    pos = wp.getpos()
+    res = wp.resolution
+    assert pos == 90*res
+
+def test_rotate_invalid_degree():
+    wp = WaveplateStub()
+    wp.connect()
+    with pytest.raises(WaveplateInvalidDegreeError):
+        wp.rotate(361)
+
+def test_disable_and_enable_channels():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.disable_channel(1)
+    wp.rotate(90)
+    # Should not move
+    pos = wp.getpos()
+    assert pos == 0
+    wp.enable_channel(1)
+    wp.rotate(90)
+    pos = wp.getpos()
+    res = wp.resolution
+    assert pos == 90*res
+
+def test_disable_invalid_channel():
+    wp = WaveplateStub()
+    wp.connect()
+    with pytest.raises(WaveplateInvalidMotorChannelError):
+        # Waveplates has max channel of 1
+        wp.disable_channel(2)
+
+def test_enable_duplicate_channel():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.disable_channel(1)
+    wp.enable_channel(1)
+    wp.enable_channel(1)
+    # Make sure there is only one channel 1
+    assert len(wp.enabled_channels) == 1
+
+def test_step_forward_backward():
+    wp = WaveplateStub()
+    wp.connect()
+    wp.rotate(90)
+    pos = wp.getpos()
+    wp.step_forward(10)
+    newPos = wp.getpos()
+    assert newPos == pos + 10
```

Comment:
Remember, camelCase variable names are not idiomatic Python. Should be `new_pos`.

```suggestion
    original_position = wp.getpos()
    wp.step_forward(10)
    assert wp.getPos() == original_position + 10
```